tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2255,3 +2265,122 @@ def do_consume():
+            topology.add_rack("region-c", Rack('az-1'))
+
+            topology.add_node("region-a", 'az-1', self.redpanda.nodes[0])
+            topology.add_node("region-b", 'az-1', self.redpanda.nodes[1])
+            topology.add_node("region-c", 'az-1', self.redpanda.nodes[2])
+
+            delay_ms = 100
+            jitter_ms = 20
+            topology.add_connection_spec(
+                TopologyConnectionSpec(spec=NetemSpec(delay_ms, jitter_ms),
+                                       source=TopologyNode(region='region-a'),
+                                       target=TopologyNode(region='region-b')))
+
+            topology.add_connection_spec(
+                TopologyConnectionSpec(spec=NetemSpec(delay_ms, jitter_ms),
+                                       source=TopologyNode(region='region-b'),
+                                       target=TopologyNode(region='region-a')))
+            topology.add_connection_spec(
+                TopologyConnectionSpec(spec=NetemSpec(delay_ms, jitter_ms),
+                                       source=TopologyNode(region='region-c'),
+                                       target=TopologyNode(region='region-a')))
+
+            topology.add_connection_spec(
+                TopologyConnectionSpec(spec=NetemSpec(delay_ms, jitter_ms),
+                                       source=TopologyNode(region='region-c'),
+                                       target=TopologyNode(region='region-b')))
+
+            topology.enable_traffic_shaping()
+
+            topology.start_with_topology(self.redpanda)
+            self.topics = [TopicSpec(partition_count=6, replication_factor=3)]
+            self._create_initial_topics()
+
+            timeout_sec = 30
+            self._start_one_producer(self.topics[0], timeout_sec=timeout_sec)
+
+            group_id = f"pandaproxy-group-{uuid.uuid4()}"
+            self._start_one_consumer(self.topics[0],
+                                     group_id,
+                                     timeout_sec=timeout_sec)
+
+            admin = Admin(self.redpanda)
+
+            brokers = admin.get_brokers()
+
+            for b in brokers:
+                n = self.redpanda.get_node_by_id(b['node_id'])
+                placement = topology.placement_of(n)
+                # TODO: change when we add support for regions in redpanda
+                assert b['rack'] == f"{placement.region}.{placement.rack}"
+
+            for prod in self.producers:
+                prod.join()
+                if prod.exception is not None:
+                    raise prod.exception
+
+            for cons in self.consumers:
+                cons.join()
+                if cons.exception is not None:
+                    raise cons.exception
+
+            # There should be many logs from transport about an RPC timeout of delay ms.
+            # There should be atleast 2 copies of the timeout because of client retry
+            pattern = f"RPC timeout ({delay_ms} ms)"
```

Comment:
It's not immediately obvious to me that this is checking that the client is retrying.

Reply:
We are delaying network reconnect tests to another time. Ticket for this here 
https://github.com/redpanda-data/redpanda/issues/10067